### ISSUE TITLE
Does this library prevent against CSV injection?

### ISSUE THREAD

**Issue by maxp-edcast** [react-csv/react-csv#156]
My test is to create a piece of content with text

```
=IMAGE("http://strabo.com/gallery/albums/wallpaper/foo_wallpaper.sized.jpg")
```

Then export some table containing this to CSV and open it in Google Sheets. 

In Google Sheets I get a parse error for this cell. If I look at the text content of the cell, it is:

```
=IMAGE(http://strabo.com/gallery/albums/wallpaper/foo_wallpaper.sized.jpg")"
```

The first `"` (before `http`) has been removed and one has been added after the final `)`

What is the actual logic being applied here? Is this done for the purpose of preventing injection, or is that just an accidental by product?

Note that I don't think causing a parse error is the right result here, ideally it would be prepending a `'` single quote before any cell content that begins with one of `=+=@`

The reason I ask this was that I added my own sanitization code using the aformentioned formula, but a coworker suggested that possibly this was already handled by the library we're using (react-csv). So I ask here, is that the case? Should we be manually escaping our CSV cells?

**Comment by vazitron** [comment]
I'm not a maintainer, but by the looks of the code, it appears that:
1. It doesn't handle CSV injection
2. It doesn't handle escaping of `separator char` and `encloser char`

I think you problem might be the fact that your value contains the encloser char. 
Try to escape it first, i.e. replace `"` with `""`,
and then inspect your result.

**Comment by mccabemj** [comment]
@maxp-edcast does the raw text of the CSV file remove that first quote? (before http)

**Comment by martinmckenna** [comment]
Here's my TypeScript solution to cleaning the data FWIW:

```js
export const cleanCSVData = (data: any): any => {
  if (data === null) {
    return null;
  }

   if (Array.isArray(data)) {
    return data.map(eachValue => {
      return cleanCSVData(eachValue);
    });
  }

   if (typeof data === 'object') {
    return Object.keys(data).reduce((acc, eachKey) => {
      acc[eachKey] = cleanCSVData(data[eachKey]);
      return acc;
    }, {});
  }

   if (typeof data === 'boolean' || typeof data === 'number') {
    return data;
  }

   /**
   * fairly confident this should be typecast as a string by now
   * basically, just get rid of operator symbols
   */
  return (data as string).replace(/[-|+|=]/g, match => `"${match}"`);
};
```
